tests/lcpcategory/test-currentCategory.php

```diff
@@ -77,6 +77,19 @@ public function test_single_post_page() {
     $this->assertSame("${cat_ID_1},${cat_ID_2}", $lcpcategory->current_category('yes'));
   }
 
+  public function test_post_with_excluded_cats() {
+    $lcpcategory = LcpCategory::get_instance();
+
+    $this->go_to('/?p=' . self::$test_post_2);
+    $this->assertQueryTrue('is_singular', 'is_single');
+    // Should produce a string with comma separated IDs, with '-' sign where
+    // necessary.
+    $this->assertSame(
+      self::$test_cat . ',-' . self::$test_cat_2,
+      $lcpcategory->current_category('yes', (string) self::$test_cat_2)
+    );
+  }
+
```

Comment:
@picandocodigo I recommend you do it this way, just returning a string with excluded categories: `1,2-3-4` is the simplest solution that works with the rest of the code in this Class. This way you don't have to work with the 'exclude' array which I think is only necessary for the 'and' relationship.

Or do you want to allow excluding IDs when `categorypage=all` as well? What do you think?